tests/import_proxy_test.go

```diff
@@ -0,0 +1,319 @@
+				config.ImportProxy = &cdiv1.ImportProxy{
+					HTTPProxy:  &http,
+					HTTPSProxy: &https,
+					NoProxy:    &noProxy,
+				}
+			})
+			Expect(err).ToNot(HaveOccurred())
+			return true
+		}
+		return false
+	}, time.Second*30, time.Second).Should(BeTrue())
+}
+
+// verifyImporterPodInfoInProxyLogs verifiy if the importer pod request (method, url and impoter pod IP) appears in the proxy log
+func verifyImporterPodInfoInProxyLogs(f *framework.Framework, dataVolume *cdiv1.DataVolume, imgURL string, isHTTPS bool, expected func() types.GomegaMatcher) {
+	podIP := getImporterPodIP(f)
+	Eventually(func() bool {
+		proxyLOG := getProxyLog(f)
+		return wasPodProxied(imgURL, podIP, proxyLOG, isHTTPS)
+	}, time.Second*60, time.Second).Should(expected())
+}
+
+func getImporterPodIP(f *framework.Framework) string {
+	podIP := ""
+	Eventually(func() string {
+		pod, err := utils.FindPodByPrefix(f.K8sClient, f.Namespace.Name, common.ImporterPodName, common.CDILabelSelector)
+		if err != nil || pod.Status.Phase == corev1.PodPending {
+			return ""
+		}
+		fmt.Fprintf(GinkgoWriter, "INFO: Checking POD %s IP: %s\n", pod.Name, pod.Status.PodIP)
+		podIP = pod.Status.PodIP
+		return podIP
+	}, time.Second*60, time.Millisecond*100).Should(Not(BeEmpty()))
+	return podIP
+}
+
+func getProxyLog(f *framework.Framework) string {
+	proxyPod, err := utils.FindPodByPrefix(f.K8sClient, f.CdiInstallNs, proxyServerName, fmt.Sprintf("name=%s", proxyServerName))
+	Expect(err).ToNot(HaveOccurred())
+	fmt.Fprintf(GinkgoWriter, "INFO: Analyzing the proxy pod %s logs\n", proxyPod.Name)
+	log, err := RunKubectlCommand(f, "logs", proxyPod.Name, "-n", proxyPod.Namespace, "--since=15m")
+	Expect(err).To(BeNil())
+	return log
+}
+
+func wasPodProxied(imgURL, podIP, proxyLog string, isHTTPS bool) bool {
+	method := "METHOD:GET"
+	if isHTTPS {
+		//Since the proxy hijacks the HTTPS conections (i.e., tunneling), the first https request uses the method "CONNECT" instead of "GET"
+		method = "METHOD:CONNECT"
+	}
+	u, _ := url.Parse(imgURL)
+	for _, line := range strings.Split(strings.TrimSuffix(proxyLog, "\n"), "\n") {
+		if strings.Contains(line, method) && strings.Contains(line, u.Host) && strings.Contains(line, podIP) {
+			fmt.Fprintf(GinkgoWriter, "INFO: Proxy log: %s\n", line)
+			fmt.Fprintf(GinkgoWriter, "INFO: The import POD requests were proxied\n")
+			return true
+		}
+	}
+	fmt.Fprintf(GinkgoWriter, "INFO: The import POD requests were not proxied\n")
+	return false
+}
+
+func cleanClusterWideProxy(f *framework.Framework) {
+	if utils.IsOpenshift(f.K8sClient) {
+		By("Restoring cluster-wide proxy object to original state")
+		ocpClient, err := configclient.NewForConfig(f.RestConfig)
+		Expect(err).ToNot(HaveOccurred())
+		proxy, err := ocpClient.ConfigV1().Proxies().Get(context.TODO(), controller.ClusterWideProxyName, metav1.GetOptions{})
+		if k8serrors.IsNotFound(err) {
+			Skip("This OpenShift cluster version does not have a Cluster Wide Proxy object")
```

Comment:
Skip is a function to tell the ginkgo framework to skip the test. The cleanClusterWideProxy is a cleanup function, which means the test has already completed. So I would just return instead of calling skip.